meteor/client/lib/triggers/TriggersHandler.tsx

```diff
@@ -163,14 +177,24 @@ export interface MountedGenericTrigger {
 	_rank: number
 	/** The ID of the action that will be triggered */
 	triggeredActionId: TriggeredActionId
-	/** Keys that have a listener mounted to */
+	/** Keys or combos that have a listener mounted to */
 	keys: string[]
+	/** Final keys in the combos */
+	finalKeys: string[]
 	/** A label of the action, if available */
 	name: string | ITranslatableMessage
+	/** Hint that all actions of this trigger are adLibs */
+	adLibOnly: boolean
 }
 
 export const MountedGenericTriggers = new Mongo.Collection<MountedGenericTrigger>(null)
 
+export function isMountedAdLibTrigger(
+	mountedTrigger: MountedAdLibTrigger | MountedGenericTrigger
+): mountedTrigger is MountedAdLibTrigger {
+	return !!mountedTrigger['targetId']
```

Comment:
`MountedAdLibTrigger`  and `MountedGenericTrigger` are not extensions of a shared base, they come from two different collections. It's not possible to mix them up, at least not here - why is this function here then? The check of `!!mountedTrigger['tragetId']` can't actually guarantee that this is an `MountedAdLibTrigger`; if the topology of `MountedAdLibTrigger` changes, this will not show up in any compilation message.

Reply:
This function also appears to be unused

Reply:
It is used by TV 2 in the code excluded from contribution, where we need to tell apart `MountedAdLibTrigger`s from `MountedGenericTrigger`s. 
you do mix them together sometimes, for example in `mountedTriggerToHotkeyList`.
Should I make them share a base class then?

Reply:
@ianshade Okay, so let's put this into a lib file or just put it next to where they are combined. Alternatively, let's refactor this so that they have a shared base and a clear discriminator for one or the other.

Reply:
What I'm specifically getting at, is that the triggers ecosystem was specifically designed to avoid using type predicates and allow the interface typings to implicitly state the types of various objects and available properties. Type predication is a huge problem with the whole `IAdLib*Ui` environment that I would like to clean up at some point: the predicates aren't very good (as they generally aren't type-safe themselves) and we use them more as a tool to make the compiler happy rather than providing actual type safety.